crates/core/src/kvs/export.rs

```diff
@@ -178,6 +156,55 @@ impl TryFrom<&Value> for TableConfig {
 	}
 }
 
+impl SurrealValue for TableConfig {
+	fn kind_of() -> Kind {
+		Kind::Object
```

Comment:
I know it's not highly important here but ideally this becomes an either type. Perhaps for these custom implementations which users will have aswell it could be nice to have a macro which can generate a kind for you based on surrealql syntax 🤔

Reply:
Oh yeah, this is wrong. It should match the assertions of `is_value` below.

I definitely would like to keep expanding the functionality of the `SurrealValue` derive macro to support enums and as much customization as we need.